src/classes/ADDR_GoogleGeoAPI_Validator.cls

```diff
@@ -99,10 +99,23 @@ public with sharing class ADDR_GoogleGeoAPI_Validator implements ADDR_IValidator
     private Address__c verifyAddress(Address__c a){
 
         String address_request_string = '';
-        if (a.MailingStreet__c != null)
-            address_request_string += a.MailingStreet__c.replace(' ', '+') + '+';
-        if (a.MailingStreet2__c != null)
-            address_request_string += a.MailingStreet2__c.replace(' ', '+') + '+';
+        String mailingStreet = '';
+        String subpremise = '';
+        
+        if (!String.isBlank(a.MailingStreet__c))
+            mailingStreet += a.MailingStreet__c.trim();
+        if (!String.isBlank(a.MailingStreet2__c))
+            mailingStreet += ' ' + a.MailingStreet2__c.trim();
+
+        //if our address has a comma, split it off and append it after verification
+        if (mailingStreet.contains(',')) {
+            list<string> split = mailingStreet.split(',',2);
+            address_request_string += 'search_address=' + split[0].replace(' ', '+') + '&';
```

Comment:
Did you manually test this works? I'm not sure Google takes ampersands, or only plus signs as separators.